### PR TITLE
layers: Add command_validation for vkCmds

### PR DESCRIPTION
**Pull request by sfricke-samsung**
First part of solution for #2720  - This change is just to migrate what is already in the layers, a second PR will be to add the missing implicit VU generation which will require removing all the `ValidateCmdQueueFlags`, `InsideRenderPass`, and `OutsideRenderPass` where possible and have them all funnel through `ValidateCmd()`

This change
- Moves the `CMD_TYPE` generation from `vk_validation_error_messages.h` to the new `command_validation.h`
    - This turns the `vk_validation_error_messages` file back to what I feel it only should have in it
    - Also allow the `vk_validation_stats.py` to properly scan the new generated file
- Moves `ValidateCmd` to `command_validation.cpp` where it will be able to handle all the auto generated implicit `vkCmd*` calls
- Make `vkEndCommandBuffer` use explicit checks like `vkBeginCommandBuffer` does

### CODE REVIEW

File: layers/generated/command_validation.cpp [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+ * limitations under the License.
+ *
+ * Author: Spencer Fricke <s.fricke@samsung.com>
+ *
+ ****************************************************************************/
+
+#include "vk_layer_logging.h"
+#include "core_validation.h"
+
+static const std::array<const char *, CMD_RANGE_SIZE> KGeneratedMustBeRecordingList = {{
```
Comment (ncesario-lunarg) [inline layers/generated/command_validation.cpp:28]: For consistency, I think this should start with a lower case `k` (i.e., `kGeneratedMustBeRecordingList`).

**Review by ncesario-lunarg — APPROVED** [review]
Aside from a single character nit, LGTM!

I added @mikes-lunarg in case he has a chance to look at the python code, but that all LGTM too.

File: scripts/lvl_genvk.py [hunk 2]
```diff
@@ -906,6 +906,59 @@ def makeGenOpts(args):
...
+            emitExtensions    = emitExtensionsPat,
+            emitSpirv         = None,
+            prefixText        = prefixStrings + vkPrefixStrings,
+            protectFeature    = False,
+            apicall           = 'VKAPI_ATTR ',
+            apientry          = 'VKAPI_CALL ',
+            apientryp         = 'VKAPI_PTR *',
+            alignFuncParam    = 48,
+            expandEnumerants  = False,
+            valid_usage_path  = args.scripts)
```
Comment (mikes-lunarg) [inline scripts/lvl_genvk.py:933]: Clever!

**Review by mikes-lunarg — APPROVED** [review]
Awesome, this was long overdue

**Comment by mark-lunarg** [comment]
Wanted to do something like this forever (but I suck at python) -- great!

**Pull request merged**